File: tests/exercises/objects/solution/correct.java
import java.util.*;

class EqualChecker {

    private final int number;
    public int prop = 0;

    EqualChecker(int number) {
        this.number = number;
    }

    public boolean check(int other) {
        return this.number == other;
    }

    public static Set<List<Integer>> setTest() {
        return Set.of(List.of(1, 2), List.of(2, 3));
    }
}

File: tests/exercises/objects/solution/correct.cpp
#include <iostream>
#include <set>
#include <vector>

class EqualChecker {
public:
    EqualChecker(int number) : number(number) {}

    bool check(int other) {
        return other == number;
    }

    int prop;
private:
    int number;
};

std::set<std::vector<int>> setTest() {
    std::set<std::vector<int>> mySet;
    mySet.insert({1, 2});
    mySet.insert({2, 3});
    return mySet;
}
